docs/reference/contributing/target/QuadSPI.md

```diff
@@ -53,4 +75,8 @@ To enable the QSPI HAL, define `QSPI` in the targets.json file inside `device_ha
 
 ### Testing
 
-To be implemented
+The Mbed OS HAL provides a set of conformance tests for the QSPI interface (**Note: QSPI HAL tests require onboard QSPI flash memory**). You can use these tests to validate the correctness of your implementation. To run the QSPI HAL tests, use the following command:
```

Comment:
> **Note: QSPI HAL tests require onboard QSPI flash memory**

I would say it's still a valid use case to support QSPI even if the board doesn't have on board qspi flash. Could we only run the test requiring onboard flash if the QSPI_FLASH1_xxx are defined? @donatieng @jamesbeyond

Reply:
@bulislaw Actually QSPI test is run only on targets defining `DEVICE_QSPI`.
Currently `DEVICE_QSPI` means device with onboard flash, so we don't recognize QSPI capable devices without onboard QSPI flash. I think that it would be good to have something like: `DEVICE_QSPI` and `DEVICE_QSPI_FLASH` to distinguish them.

Reply:
Why do we only consider devices with onboard flash? Not every SPI device has SPI memory attached to it...

Reply:
Not every. 
e.g. Nucleo-L496ZG  has no QSPI flash but it has QSPI signals routed out on external connector according schematics (waiting for STM confirmation on it in https://github.com/ARMmbed/mbed-os/pull/7817)

Reply:
> I would say it's still a valid use case to support QSPI even if the board doesn't have on board qspi flash. Could we only run the test requiring onboard flash if the QSPI_FLASH1_xxx are defined? @donatieng @jamesbeyond

👍  Wording would be "requires QSPI FLASH pins to be defined" ? Targets with onboard flash would have it by default, others would need a config to add them.